Path: news.gmane.org!not-for-mail
From: Anton Blanchard <anton@samba.org>
Newsgroups: gmane.linux.network
Subject: [PATCH 3/4] ibmveth: Checksum offload is always disabled
Date: Thu, 08 Sep 2011 10:41:05 +1000
Lines: 23
Approved: news@gmane.org
Message-ID: <20110908004121.744925909@samba.org>
References: <20110908004102.355674129@samba.org>
NNTP-Posting-Host: lo.gmane.org
X-Trace: dough.gmane.org 1315442745 17939 80.91.229.12 (8 Sep 2011 00:45:45 GMT)
X-Complaints-To: usenet@dough.gmane.org
NNTP-Posting-Date: Thu, 8 Sep 2011 00:45:45 +0000 (UTC)
Cc: netdev@vger.kernel.org
To: Santiago Leon <santil@linux.vnet.ibm.com>,
	brking@linux.vnet.ibm.com, rcj@linux.vnet.ibm.com,
	mirq-linux@rere.qmqm.pl
Original-X-From: netdev-owner@vger.kernel.org Thu Sep 08 02:45:39 2011
Return-path: <netdev-owner@vger.kernel.org>
Envelope-to: linux-netdev-2@lo.gmane.org
Original-Received: from vger.kernel.org ([209.132.180.67])
	by lo.gmane.org with esmtp (Exim 4.69)
	(envelope-from <netdev-owner@vger.kernel.org>)
	id 1R1SkN-0001Tk-2c
	for linux-netdev-2@lo.gmane.org; Thu, 08 Sep 2011 02:45:39 +0200
Original-Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1757675Ab1IHApb (ORCPT <rfc822;linux-netdev-2@m.gmane.org>);
	Wed, 7 Sep 2011 20:45:31 -0400
Original-Received: from ozlabs.org ([203.10.76.45]:45063 "EHLO ozlabs.org"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S1757657Ab1IHAp1 (ORCPT <rfc822;netdev@vger.kernel.org>);
	Wed, 7 Sep 2011 20:45:27 -0400
Original-Received: from localhost (ppp121-44-79-234.lns20.syd6.internode.on.net [121.44.79.234])
	(using TLSv1 with cipher DHE-RSA-AES128-SHA (128/128 bits))
	(Client did not present a certificate)
	by ozlabs.org (Postfix) with ESMTPSA id 17FFBB6F98;
	Thu,  8 Sep 2011 10:45:26 +1000 (EST)
X-Mailbox-Line: From anton@samba.org Thu Sep  8 10:41:21 2011
User-Agent: quilt/0.48-1
Content-Disposition: inline; filename=ibmveth_fix_csum.patch
Original-Sender: netdev-owner@vger.kernel.org
Precedence: bulk
List-ID: <netdev.vger.kernel.org>
X-Mailing-List: netdev@vger.kernel.org
Xref: news.gmane.org gmane.linux.network:205929
Archived-At: <http://permalink.gmane.org/gmane.linux.network/205929>

Commit b9367bf3ee6d (net: ibmveth: convert to hw_features) reversed
a check in ibmveth_set_csum_offload that results in checksum offload
never being enabled.

Signed-off-by: Anton Blanchard <anton@samba.org>
Cc: <stable@kernel.org> # 3.0+
---

Index: linux-build/drivers/net/ibmveth.c
===================================================================
--- linux-build.orig/drivers/net/ibmveth.c	2011-09-01 16:02:12.198726425 +1000
+++ linux-build/drivers/net/ibmveth.c	2011-09-01 16:05:37.282482851 +1000
@@ -812,7 +812,7 @@ static int ibmveth_set_csum_offload(stru
 		} else
 			adapter->fw_ipv6_csum_support = data;
 
-		if (ret != H_SUCCESS || ret6 != H_SUCCESS)
+		if (ret == H_SUCCESS || ret6 == H_SUCCESS)
 			adapter->rx_csum = data;
 		else
 			rc1 = -EIO;


